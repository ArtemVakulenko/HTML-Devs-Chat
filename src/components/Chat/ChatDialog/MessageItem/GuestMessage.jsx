import React from 'react';
import PropTypes from 'prop-types';
import { useTheme } from 'src/components/hooks';
import Button from 'src/components/_common/Button';
import { icons } from 'constants';
import { 
    StGuestMessageContainer, 
    StGuestMessageItem,
    StGuestName,
    StGuestMessageContent,
    StDate,
   } from './styled';

const GuestMessage = ({ authorPic, authorName, content, date, onClick }) => {
    const { colors, theme } = useTheme();
    return (
        <StGuestMessageContainer>
        <Button icon={authorPic || icons.defaultUserIcon} onClick={onClick}/>
        <StGuestMessageItem colors={colors} theme={theme}>
          <StGuestName colors={colors} theme={theme}>
            {authorName}
          </StGuestName>
          <StGuestMessageContent colors={colors} theme={theme}>
            {content}
          </StGuestMessageContent>
          <StDate colors={colors} theme={theme}>
            {date}
          </StDate>
        </StGuestMessageItem>
        </StGuestMessageContainer>
    );
};

GuestMessage.propTypes = {
    authorPic: PropTypes.string,
    authorName: PropTypes.string,
    content: PropTypes.string,
    date: PropTypes.string,
    onClick: PropTypes.func,
};

export default GuestMessage;
