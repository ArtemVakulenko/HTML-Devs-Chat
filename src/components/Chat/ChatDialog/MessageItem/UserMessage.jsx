import React from 'react';
import PropTypes from 'prop-types';
import { useTheme } from 'src/components/hooks';
import Button from 'src/components/_common/Button';
import { icons } from 'constants';
import {
  StUserMessageContainer,
  StUserMessageItem,
  StUserMessageContent,
  StDate,
} from './styled';

const UserMessage = ({ authorPic, content, date, onClick }) => {
  const { colors, theme } = useTheme();
  return (
    <StUserMessageContainer >
      <StUserMessageItem colors={colors} theme={theme}>
        <StUserMessageContent colors={colors} theme={theme}>
          {content}
        </StUserMessageContent>
        <StDate colors={colors} theme={theme}>
          {date}
        </StDate>
      </StUserMessageItem>
      <Button icon={authorPic || icons.defaultUserIcon} onClick={onClick} />
    </StUserMessageContainer>
  );
};

UserMessage.propTypes = {
  authorPic: PropTypes.string,
  content: PropTypes.string,
  date: PropTypes.string,
  onClick: PropTypes.func,
};

export default UserMessage;
