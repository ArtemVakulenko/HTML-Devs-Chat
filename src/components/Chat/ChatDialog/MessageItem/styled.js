import styled from 'styled-components';

export const StUserMessageContainer = styled.div`
    display: flex;
    justify-content: flex-end;
    align-items: center;
    min-height: 15%;
    padding: 10px;
    height: auto;
`;

export const StGuestMessageContainer = styled.div`
    display: flex;
    justify-content: flex-start;
    align-items: center;
    min-height: 15%;
    padding: 10px;
`;

export const StUserMessageItem = styled.div`
    background-color: ${({ colors, theme }) => colors[theme].message};
    display: flex;
    border-radius: 10px;
    max-width: 50%;
    min-width: 100px;
    flex-direction: column;
    height: fit-content;
    padding: 5px 10px;
`;

export const StGuestMessageItem = styled.div`
    background-color: ${({ colors, theme }) => colors[theme].secondary};
    border-radius: 10px;
    display: flex;
    max-width: 50%;
    min-width: 100px;
    flex-direction: column;
    height: fit-content;
    padding: 5px 10px;
`;

export const StGuestName = styled.div`
    color: ${({ colors, theme }) => colors[theme].guestName};
    font-size: 1.1rem;
    font-weight: bold;
`;

export const StUserMessageContent = styled.div`
    color: ${({ colors, theme }) => colors[theme].font};
    margin-top: 5px;
    word-wrap: break-word;
`;

export const StGuestMessageContent = styled.div`
    color: ${({ colors, theme }) => colors[theme].color};
    margin-top: 5px;
    word-wrap: break-word;
`;

export const StDate = styled.div`
    color: grey;
    display: flex;
    justify-content: flex-end;
    padding: 3px;
    margin: 3px;
    font-size: 0.6rem;
`;
