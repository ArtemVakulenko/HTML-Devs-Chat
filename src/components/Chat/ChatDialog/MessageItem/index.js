import { userName, getUserPic } from 'store/mainPage/selectors';
import { connect } from 'react-redux';
import { getOneUser } from 'store/modals/actions';
import MessageItem from './MessageItem';

const mapStateToProps = (state) => ({
  userName: userName(state),
  userPic: getUserPic(state),
});

const mapDispatchToProps = {
  getOneUser,
};
export default connect(mapStateToProps, mapDispatchToProps)(MessageItem);
