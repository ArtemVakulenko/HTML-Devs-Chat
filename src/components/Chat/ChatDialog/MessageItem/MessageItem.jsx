import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { getTimeString } from 'src/helpers';
import UserMessage from './UserMessage';
import GuestMessage from './GuestMessage';
import ShowUserInfoModal from '../ShowUserInfoModal';

const MessageItem = ({
  date,
  author,
  userPic,
  content,
  userName,
  authorPic,
  getOneUser,
   }) => {
  const msgDate = getTimeString(date);
  const [isVisible, setVisible] = useState(false);

  const modalOn = () => {
    setVisible(true);
    getOneUser(author);
  };
  const modalOff = () => {
    setVisible(false);
  };
    return (
      <>
      {
        userName === author
        ? (
          <UserMessage
            date={msgDate}
            content={content}
            authorPic={userPic}
            onClick={modalOn}
          />
        ) : (
          <GuestMessage
          date={msgDate}
          content={content}
          authorPic={authorPic}
          authorName={author}
          onClick={modalOn}
          />
        )
      }
      {isVisible
      ? <ShowUserInfoModal onCloseModal={modalOff} login={author}/>
      : null}
      </>
    );
};

MessageItem.propTypes = {
  date: PropTypes.number,
  author: PropTypes.string,
  content: PropTypes.string,
  userPic: PropTypes.string,
  userName: PropTypes.string,
  authorPic: PropTypes.string,
  getOneUser: PropTypes.func,
};

export default MessageItem;
