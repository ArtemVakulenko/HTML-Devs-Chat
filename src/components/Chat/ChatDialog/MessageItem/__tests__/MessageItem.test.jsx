import React from 'react';
import configureStore from 'redux-mock-store';
import MessageItem from '../MessageItem';
import { shallowSmart } from '../../../../../__tests__/testHelper';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('../../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));
const mockStore = configureStore();
const store = mockStore({ mainPage: {
  sidebar: {
    search: '',
    rooms: [],
  },
  chat: {
    messages: [],
    roomId: '',
    messageValue: '',
    activeRoom: {},
  },
  user: {
    id: '',
    login: '',
    name: '',
    lastName: '',
    img: '',
    hobbie: '',
    city: '',
    age: '',
  },
} });

jest.mock('react-notifications', () => ({
  NotificationManager: {
    error: jest.fn(),
    success: jest.fn(),
  },
}));
describe('MessageItem', () => {
  describe('123', () => {
    const props = {
      message: {},
    };
    it('Should match snapshot', () => {
      const component = shallowSmart(<MessageItem {...props} />, store);
      expect(component.html()).toMatchSnapshot();
    });
  });
});
