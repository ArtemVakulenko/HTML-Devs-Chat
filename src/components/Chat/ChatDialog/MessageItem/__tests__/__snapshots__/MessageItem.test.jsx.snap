// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage 123 Should match snapshot 1`] = `"<div class=\\"styled__StUserMessageContainer-kRKSqR dLyyDN\\"><div class=\\"styled__StUserMessageItem-iHlOmP jztccz\\"><div class=\\"styled__StUserMessageContent-hxJINz grdRaV\\"></div><div class=\\"styled__StDate-dCFtOY fvsCbz\\">NaN.NaN NaN:NaN</div></div><button class=\\"styled__StIcon-chPeWD bvoJqX\\"></button></div>"`;

exports[`MessageItem 123 Should match snapshot 1`] = `"<div class=\\"styled__StUserMessageContainer-kRKSqR dLyyDN\\"><div class=\\"styled__StUserMessageItem-iHlOmP jztccz\\"><div class=\\"styled__StUserMessageContent-hxJINz grdRaV\\"></div><div class=\\"styled__StDate-dCFtOY fvsCbz\\">NaN.NaN NaN:NaN</div></div><button class=\\"styled__StIcon-chPeWD bvoJqX\\"></button></div>"`;
